src/login/logind-session.c

```diff
@@ -180,14 +191,17 @@ int session_set_leader_consume(Session *s, PidRef _leader) {
         if (pidref_equal(&s->leader, &pidref))
                 return 0;
 
-        r = hashmap_put(s->manager->sessions_by_leader, PID_TO_PTR(pidref.pid), s);
+        r = pidref_dup(&pidref, &pid_dup);
```

Comment:
Why not to directly use `s->leader`?
I know `pidref_hash_ops` frees the PidRef, but I think in this case it is better to rename `pidref_hash_ops` -> `pidref_hash_ops_free`, and re-introduce `pidref_hash_ops` without destructor, and use the new hash ops here.